clients/drcachesim/tools/schedule_stats.cpp

```diff
@@ -271,10 +271,16 @@ schedule_stats_t::parallel_shard_memref(void *shard_data, const memref_t &memref
         line << "Core #" << std::setw(2) << shard->core << " @" << std::setw(9)
              << shard->stream->get_record_ordinal() << " refs, " << std::setw(9)
              << shard->stream->get_instruction_ordinal() << " instrs: input "
-             << std::setw(4) << input_id << " @" << std::setw(9)
-             << shard->stream->get_input_interface()->get_record_ordinal() << " refs, "
+             << std::setw(4) << input_id << " @"
              << std::setw(9)
-             << shard->stream->get_input_interface()->get_instruction_ordinal()
+             // The interface is null when we see idle records.
+             << (shard->stream->get_input_interface() == nullptr
+                     ? -1
+                     : shard->stream->get_input_interface()->get_record_ordinal())
+             << " refs, " << std::setw(9)
+             << (shard->stream->get_input_interface() == nullptr
+                     ? -1
+                     : shard->stream->get_input_interface()->get_instruction_ordinal())
```

Comment:
nit: maybe we should save `shard->stream->get_input_interface()` to a local variable and use that in this case and the three other places above.
Unrelated, but `input_id` at L268 looks like it's used only in "verbose mode", maybe it should be scoped inside this if.

Reply:
Done on the local var.  input_id is actually used further down below in this function.

Reply:
Ah, you're right, I didn't "expand" down enough.